### ISSUE TITLE
Bug when try to fine-tune the pretrained models.

### ISSUE THREAD

**Issue by Geo99pro** [JulianGuerreiro/LayoutFlow#2]
Hi @JulianGuerreiro 

I hope you are doing well.

First, I would like to congratulate you and your research team on your paper LayoutFlow: Flow Matching for Layout Generation.

I successfully ran the source code available. Currently, I am trying to fine-tune the pretrained model you shared on  [Hugging Face repository](https://huggingface.co/JulianGuerreiro/LayoutFlow) using a different dataset that I have formatted in the h5py format.

However, I noticed that when I start training, the model stays in evaluation mode, which seems unusual. I added a callback following the  [pytorch_lighting doc](https://pytorch-lightning.readthedocs.io/en/1.1.8/generated/pytorch_lightning.callbacks.Callback.html#pytorch_lightning.callbacks.base.Callback.on_fit_start:~:text=on_fit_start(trainer%2C%20pl_module)) , which allowed me to switch it back to training mode.

But by doing this, I observed that the FID model also began training.

Could you please let me know if I might be doing something wrong ?

**Before the callback** 
![image](https://github.com/user-attachments/assets/dfe3c4d3-ee7f-436a-ade6-d7d7a8ac8af2)

**After the callback**
![image](https://github.com/user-attachments/assets/5b637211-96ed-4f5f-a059-c4d6a0256ee6)

**Comment by JulianGuerreiro** [comment]
Hi, thank you for your message! 

Your problem does indeed sound strange, as I've never really encountered that issue. 
I'm assuming you are using the `train.py` script for training, but could you maybe share which configurations you are running the script with?

Usually, PyTorch Lighting should be handling the training/eval mode stuff. While it would be better to have the fid model in eval mode (for things like dropout etc.), the screenshot you shared actually shows that the 2.9M parameters of the fid model are *non*-trainable params. So you are not updating the parameters of the FID model. This is because in the [optimizer function](https://github.com/JulianGuerreiro/LayoutFlow/blob/b19cb8510d249460b00e892cf83fb321f81b545f/src/models/BaseGenModel.py#L40]) I only register the parameters of the backbone model and never registered the FID model for optimization.

**Comment by Geo99pro** [comment]
Hi Sir @JulianGuerreiro,

I really appreciate your response.
You are absolutely right; I was trying to do something that is no longer necessary because your current code is well-structured. And yes i'm using the ``train.py`` script.

While debugging the code line by line, I noticed that the FID model is not being trained.
Only the LayoutFlow model has its parameters updated.
